src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stageState = [];
+  const stateClone = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateClone, obj.extraData);
+    }
+
+    if (obj.type === 'clear') {
+      for (const key in stateClone) {
+        delete stateClone[key];
+      }
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        delete stateClone[key];
+      }
+    }
```

Comment:
```suggestion
    switch (obj.type) {
      case 'addProperties' :
          Object.assign(stateClone, obj.extraData);
          break;
       case 'clear': 
          for (const key in stateClone) {
            delete stateClone[key];
          }
          break;
       case 'removeProperties' : 
          for (const key of obj.keysToRemove) {
            delete stateClone[key];
          }
   }  
```
You could change if construction to switch